File: backend/src/main/java/org/multimes/backend/core/web/repository/MessageRepository.java
package org.multimes.backend.core.web.repository;

import org.multimes.backend.core.web.model.entities.Message;
import org.multimes.backend.core.web.repository.interfaces.IMessageRepository;
import org.springframework.jdbc.core.JdbcTemplate;
import org.springframework.stereotype.Repository;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

@Repository
public class MessageRepository implements IMessageRepository {
    private final JdbcTemplate jdbcTemplate;

    public MessageRepository(JdbcTemplate jdbcTemplate) {
        this.jdbcTemplate = jdbcTemplate;
    }

    @Override
    public void add(Message message) {
        String sql = "insert into messages (mes_text, mes_time, is_inter, inter_id) values (?, ?, ?, ?)";
        jdbcTemplate.update(sql,
                message.getMesText(), message.getMesTime(), message.getIsInter(), message.getInterId());
    }

    @Override
    public List<Message> getAllByInterId(int id) {
        String sql = "select * from messages where inter_id = ?";
        List<Map<String, Object>> rows = jdbcTemplate.queryForList(sql, id);
        List<Message> result = new ArrayList<>();
        for (Map<String, Object> row : rows) {
            Message message = new Message(
                    (Integer) row.get("mes_id"),
                    (String) row.get("mes_text"),
                    (String) row.get("mes_time"),
                    (Boolean) row.get("is_inter"),
                    (Integer) row.get("inter_id"));
            result.add(message);
        }
        return result;
    }
}

File: backend/src/main/kotlin/org/multimes/backend/core/web/repository/MessageRepository.kt
package org.multimes.backend.core.web.repository

import org.multimes.backend.core.web.model.entities.Message
import org.multimes.backend.core.web.repository.interfaces.IMessageRepository
import org.springframework.jdbc.core.JdbcTemplate
import org.springframework.stereotype.Repository

@Repository
class MessageRepository(private val jdbcTemplate: JdbcTemplate) : IMessageRepository {

    override fun add(message: Message) {
        val sql = "insert into messages (mes_text, mes_time, is_inter, inter_id) values (?, ?, ?, ?)"
        jdbcTemplate.update(sql, message.mesText, message.mesTime, message.isInter, message.interId)
    }

    override fun getAllByInterId(id: Int): List<Message> {
        val sql = "select * from messages where inter_id = ?"
        val rows = jdbcTemplate.queryForList(sql, id)
        return rows.map { row ->
            Message(
                mesId = row["mes_id"] as Int,
                mesText = row["mes_text"] as String?,
                mesTime = row["mes_time"] as String?,
                isInter = row["is_inter"] as Boolean,
                interId = row["inter_id"] as Int
            )
        }
    }
}
